### PR TITLE
Update to typed-protocols 0.3.0.0

### PR DESCRIPTION
**Pull request by Icelandjack**
Update `ekg-forward` to the latest version of `typed-protocols-0.3`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.6.1 - Oct 2024
```
Comment (mgmeier) [inline CHANGELOG.md:3]: Can we bump to 0.7 please? Makes integration upstream much easier.

File: ekg-forward.cabal [hunk 2]
```diff
@@ -65,12 +65,13 @@ library
                        , io-classes >= 1.4.1
```
Comment (mgmeier) [inline ekg-forward.cabal:65]: please bump version in cabal file

**Review by mgmeier — CHANGES_REQUESTED** [review]

File: src/System/Metrics/Protocol/Forwarder.hs [hunk 3]
```diff
@@ -21,33 +22,32 @@ import           System.Metrics.Protocol.Type
...
-      Await (ClientAgency TokIdle) $ \case
-        -- The acceptor sent us a request for new metrics, so now we're
-        -- in the 'StBusy' state which means it's the forwarder's turn to send
-        -- a reply.
-        MsgReq req -> Effect $ do
+      Await \case
+        MsgReq req -> Effect do
           resp <- recvMsgReq req
-          return $ Yield (ServerAgency TokBusy) (MsgResp resp) go
+          return do Yield (MsgResp resp) go
```
Comment (coot) [inline src/System/Metrics/Protocol/Forwarder.hs:49]: Using `do` as `$` is rather extravagant :grin:.

**Review by coot — APPROVED** [review]
LGTM

File: src/System/Metrics/Protocol/Forwarder.hs (hunk 3 above)
Comment (Icelandjack) [reply src/System/Metrics/Protocol/Forwarder.hs:49]: Yeah I didn't like it actually, I have gone too far this time!

**Review by mgmeier — APPROVED** [review]
Thank you @Icelandjack

**Pull request merged**